Simcoe/Simcoe.swift

```diff
@@ -105,8 +105,9 @@ public final class Simcoe {
     /// - Parameters:
     ///   - product: The SimcoeProductConvertible instance.
     ///   - eventProperties: The event properties.
-    func logAddToCart<T: SimcoeProductConvertible>(_ product: T, eventProperties: Properties?) {
-        let providers: [CartLogging] = findProviders()
+    func logAddToCart<T: SimcoeProductConvertible>(_ product: T,
+                      eventProperties: Properties?, providers: [AnalyticsTracking] = Simcoe.engine.providers) {
```

Comment:
I don't think the default argument here is quite right. It won't work for people who don't use the singleton engine instance. In that case, `Simcoe.engine.providers` would return an empty list since it's a separate instance.